controllers/specialresource_controller.go

```diff
@@ -51,7 +51,7 @@ type SpecialResourceReconciler struct {
 
 // Reconcile Reconiliation entry point
 func (r *SpecialResourceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	return ReportSpecialResourcesStatus(r, req)
+	return UpgradeSpecialResource(r, req)
```

Comment:
@zvonkok can we refactor this a little by putting the 3 steps (Upgrade, Update status, reconcile CRs) here in the Reconcile function instead of calling UpgradeSpecialResource(..) which calls ReportSpecialResourcesStatus(..) which calls ReconcilerSpecialResources(..)? Just to make it more clear what the Reconcile loop is doing for someone looking for the first time.

Reply:
@dagrayvid Done, PTAL

Reply:
Thanks, I think this is a big improvement in readability